Framework/Core/include/Framework/ASoA.h

```diff
@@ -1501,76 +1501,79 @@ void notBoundTable(const char* tableName);
+                                                                                            \
+    _Name_##IdSlice() = default;                                                            \
+    _Name_##IdSlice(_Name_##IdSlice const& other) = default;                                \
+    _Name_##IdSlice& operator=(_Name_##IdSlice const& other) = default;                     \
+    std::array<_Type_, 2> inline getIds() const                                             \
+    {                                                                                       \
+      return _Getter_##Ids();                                                               \
+    }                                                                                       \
+                                                                                            \
+    bool has_##_Getter_() const                                                             \
+    {                                                                                       \
+      auto a = *mColumnIterator;                                                            \
+      return a[0] >= 0 && a[1] >= 0;                                                        \
+    }                                                                                       \
+                                                                                            \
+    std::array<_Type_, 2> _Getter_##Ids() const                                             \
+    {                                                                                       \
+      auto a = *mColumnIterator;                                                            \
+      return std::array{a[0], a[1]};                                                        \
+    }                                                                                       \
+                                                                                            \
+    template <typename T>                                                                   \
+    auto _Getter_##_as() const                                                              \
+    {                                                                                       \
+      if (O2_BUILTIN_UNLIKELY(mBinding == nullptr)) {                                       \
+        o2::soa::notBoundTable(#_Table_);                                                   \
+      }                                                                                     \
+      if (O2_BUILTIN_UNLIKELY(!has_##_Getter_())) {                                         \
+        return static_cast<T const*>(mBinding)->emptySlice();                               \
+      }                                                                                     \
+      auto a = *mColumnIterator;                                                            \
```

Comment:
also, are you sure the compiler does not optimize this out in any case?

Reply:
`operator*` contains branching and copies several pointers, and this would only be a copy of a pair of int32s, would be also easier for the compiler to optimize.

Reply:
I haven't checked the asm, although `operator*` is indeed const.